tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -4385,19 +4385,17 @@ bool MIOpenSupport::DoPoolBackward(
   }
 
   DeviceMemory<uint8> dest2;  // duplicated dest from forward:
-  int dest2_size = 0;
+  int64 dest2_size = 0;
```

Comment:
I remember you previously had a diff that get rid of `dest2`, and used `output_data` argument directly. What's wrong with that approach that make you revert it?

Reply:
some tests were failing, I think (but that may have been caused by a different problem)

Reply:
Do you have a list of what they are, or why they are failing? It will be great if we don't have to do a forward pooling in the backward pass.

Reply:
We're going to find out in a few hours

Reply:
So, the failing tests are
//tensorflow/cc:gradients_nn_grad_test 
//tensorflow/python/kernel_tests:fractional_max_pool_op_test
//tensorflow/python:auto_mixed_precision_test
//tensorflow/python/kernel_tests:pooling_ops_3d_test
//tensorflow/python/kernel_tests:pooling_ops_test 

Looks like all tests for maxpool backpropagation are failing. (And avgpool backpropagation does not need the output buffer in the first place.)

It's odd.

Reply:
That is indeed very odd. `output_data` should be equivalent to `dest2`. If in anyway they two are not equivalent, then it means a problem in not forwarding `output_data` correctly from the forward pass.